build.sh

```diff
@@ -0,0 +1,266 @@
+        fi
+    done
+    loginfo "All requirements are ok."
+}
+
+function create_build_env {
+    logdebug "Create build environment..."
+    # Dir structure
+    [[ -d $BUILD_DIR ]] && old ./$BUILD_DIR
+    mkdir -p $BUILD_DIR
+    [[ -d $BUILD_DIR/$GEEKODOC1_PATH ]] || mkdir -p $BUILD_DIR/$GEEKODOC1_PATH
+    [[ -d $BUILD_DIR/$GEEKODOC2_PATH ]] || mkdir -p $BUILD_DIR/$GEEKODOC2_PATH
+
+    cp $GEEKODOC1_PATH/*.rnc $BUILD_DIR/$GEEKODOC1_PATH
+    cp $GEEKODOC2_PATH/*.rnc $BUILD_DIR/$GEEKODOC2_PATH
+    # but remove any flat files:
+    rm $BUILD_DIR/$GEEKODOC1_PATH/*-flat.rnc $BUILD_DIR/$GEEKODOC2_PATH/*-flat.rnc
+    # Copy DocBook5 schema:
+    cp $DOCBOOKXI_RNC_PATH $BUILD_DIR/$GEEKODOC1_PATH
+    cp $DOCBOOKXI_RNC_PATH $BUILD_DIR/$GEEKODOC2_PATH
+    logdebug "Build environment created."
+}
+
+function rnc_to_rng {
+    local files="$1"
+
+    loginfo "Convert RNC -> RNG..."
+    for f in $files; do
+       rnc="$f.rnc"
+       rng="$f.rng"
+       logdebug "Converting $rnc -> $rng"
+       trang $rnc $rng
+    done
+}
+
+function make_flat {
+    local files="$1"
+
+    loginfo "Convert RNG -> RNI (flatten)..."
+    for f in $files; do
+      rng="$f.rng"
+      rni="$f-flat.rni"
+      logdebug "Converting $rng -> $rni"
+      rnginline $rng $rni
+    done
+}
+
+
+function cleanup_xml {
+    local files="$1"
+
+    loginfo "Cleanup flat RNI..."
+    for f in $files; do
+      rng="$f-flat.rng"
+      rni="$f-flat.rni"
+      logdebug "Cleanup $rni -> $rng"
+      xmllint -o $rng --nsclean --format $rni
+    done
+}
+
+function rngflat_to_rnc {
+    local files="$1"
+
+    loginfo "Convert flat RNG -> flat RNC..."
+    for f in $files; do
+      rng="$f-flat.rng"
+      rnc="$f-flat.rnc"
+      logdebug "Cleanup $rng -> $rnc"
+      trang $rng $rnc
+    done
+}
+
+function copy_flat_rnc {
+    local files="$1"
+
+    loginfo "Copy flat RNC files..."
+    for f in $files; do
+        rnc="$f-flat.rnc"
+        target="${f#$BUILD_DIR/}"
+        target="${target%/*}"
+        logdebug "Copy $rnc -> $target"
+        sed -i 's/[[:blank:]]+$//' $rnc
+        cp $rnc $target
+    done
+}
+
+# -- CLI parsing
+ARGS=$(getopt -o h,:v,b: -l help,builddir: -n "$ME" -- "$@")
+eval set -- "$ARGS"
+while true; do
+  case "$1" in
+    --help|-h)
+        usage
+        exit 0
+        shift
+        ;;
+    -b|--builddir)
+        BUILD_DIR="$2"
+        shift 2
+        ;;
+    -v)
+        VERBOSITY=$((VERBOSITY+1))
+        shift
+        ;;
+    --) shift ; break ;;
+    *) exit_on_error "Wrong parameter: $1" ;;
+  esac
+done
+
+# Fall back to 3 (=DEBUG) if we got more than 4
+[[ $VERBOSITY -ge 3 ]] && VERBOSITY=3
+LOGGING_LEVEL=${LEVEL2LOG[$VERBOSITY]}
+
+# -- Process
+files="$BUILD_DIR/$GEEKODOC1_PATH/$GEEKODOC1_NAME \
+       $BUILD_DIR/$GEEKODOC2_PATH/$GEEKODOC2_NAME"
+requires $files
+create_build_env $files
+rnc_to_rng $files
+make_flat $files
+cleanup_xml $files
+rngflat_to_rnc $files
+copy_flat_rnc $files
+
```

Comment:
* A clean-up of the build dir might be positive, you only need geekodoc-flat.rn{c,g} but you get absolutely everything.

* sch-fix.xsl is not applied anywhere. Is it not necessary anymore?

* At least for Geekodoc v2, could you add a `geekodoc-its-transclusion-flat.rnc` (with ITS and Transclusion support) perhaps? (For now, maybe in addition to the regular one. But it might help to be able to dabble with those features.)

Reply:
> * A clean-up of the build dir at the end might be positive

I'm more in favor of doing the cleanup when the script is run for the second time. The reason for this is, sometimes I need to check/debug things in the build directory. When it's gone, that would be a bit difficult. :wink: 

Maybe we could introduce a new option for that.

> `sch-fix.xsl` is not applied anywhere. Is it not necessary anymore?

This does not affect the RNC/RNG schema. The DocBook5 schema had a bug in its Schematron patterns. This stylesheet fixes this.

> At least for Geekodoc v2, could you add a geekodoc-its-transclusion-flat.rnc [...]

I would prefer to do that in a separate PR. There are so many commits now and I would like to focus only on the building process and laying the ground for the upcoming v2 release.

Of course, if such a schema is needed, we can add it later.